File: Source/Camping Mod 2.0/common/rikmuld/camping/misc/damagesources/DamageSourceBleeding.java
package rikmuld.camping.misc.damagesources;

import java.util.Random;

import net.minecraft.entity.EntityLivingBase;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.util.ChatMessageComponent;
import net.minecraft.util.DamageSource;

public class DamageSourceBleeding extends DamageSource{

	public DamageSourceBleeding(String name) 
	{
		super(name);
	}
	
	@Override
    public ChatMessageComponent getDeathMessage(EntityLivingBase entity)
    {
        ChatMessageComponent message = new ChatMessageComponent();
    	
        Random random = new Random();
        int num = random.nextInt(5);
        
        if(entity instanceof EntityPlayer) 
        {
        	if(num==0)message.addText(((EntityPlayer)entity).username + " bled away");   
        	if(num==1)message.addText(((EntityPlayer)entity).username + " has run out of blood");        
        	if(num==2)message.addText(((EntityPlayer)entity).username + " bled out");   
        	if(num==3)message.addText(((EntityPlayer)entity).username + " bled to death");      
        	if(num==4)message.addText(((EntityPlayer)entity).username + " fizzled");             	
        }
        return message;
    }
}

File: scala/com/rikmuld/camping/features/blocks/trap/DamageSourceBleeding.scala
package com.rikmuld.camping.features.blocks.trap

import java.util.Random

import net.minecraft.entity.EntityLivingBase
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.DamageSource
import net.minecraft.util.text.{ITextComponent, TextComponentString}

object DamageSourceBleeding {
  final val random =
    new Random()
}

class DamageSourceBleeding(name: String) extends DamageSource(name) {
  override def getDeathMessage(entity: EntityLivingBase): ITextComponent =
    entity match {
      case player: EntityPlayer =>
        getMessage(player, DamageSourceBleeding.random.nextInt(5))
      case _ =>
        null
    }

  def getMessage(player: EntityPlayer, i: Int): TextComponentString =
    new TextComponentString(player.getDisplayName + " " + {
      i match {
        case 0 => "bled out"
        case 1 => "bled away"
        case 2 => "has run out of blood"
        case 3 => "bled to death"
        case 4 => "fizzled"
      }
    })
}